Gems/Atom/Feature/Common/Code/Source/ImGui/ImGuiPass.cpp

```diff
@@ -608,10 +622,21 @@ namespace AZ
                         //otherwise it is possible to have a frame where scissor bounds can be bigger than window's bounds if we resize the window
                         scissorMaxX = AZStd::min(scissorMaxX, m_viewportWidth);
                         scissorMaxY = AZStd::min(scissorMaxY, m_viewportHeight);
-                        
+
+                        // check if texture id needs to be bound to the srg
+                        uint32_t index = 0;
+                        if (drawCmd.TextureId != ImGui::GetIO().Fonts->TexID)
+                        {
+                            Data::Instance<RPI::StreamingImage> img = reinterpret_cast<RPI::StreamingImage*>(drawCmd.TextureId);
+                            // Texture index 0 is reserved for the font atlas, so we start from 1 to 15 for user textures.
+                            index = aznumeric_cast<uint32_t>(m_userTextures.size() + 1);
```

Comment:
Need to add check to prevent an edge case if there are more than 16 textures

Reply:
Added check to use only the first 15 user textures. If they go above that, then it will default to the font atlas texture.